### ISSUE TITLE
Possible NRE from built-in generator when custom UserControl contains member with [DllImport] 

### ISSUE THREAD

**Issue by BreyerW** [AvaloniaUI/Avalonia#12531]
**Describe the bug**
When porting 0.10 to 11.03 i had NRE seemingly out of nowhere from AVLN:00C on `axaml` file that wasn't `MainWindow` and very plain XAML. Even `Main` entry breakpoint wasn't hit and that gave me clue there must be something wrong with both my class and the all new generator. And sure enough my class that was complaining had something like

```csharp
[DllImport("USER32.DLL")]
public static extern bool EnumWindows(EnumWindowsProc enumFunc, IntPtr lParam);
```
From what i can tell its `[DllImport]` fault cause when i comment out the attribute, my program starts (but fails with `TypeLoadException` due to missing `DllImport` on extern method)

**To Reproduce**

1. Create simple custom `UserControl`
2. Copy-pasta `[DllImport("USER32.DLL")] public static extern bool EnumWindows(EnumWindowsProc enumFunc, IntPtr lParam);` into that control
3. Start program within VS
4. See error and even `Main` entry point doesn't start

**Expected behavior**
I would expect generator for avalonia to ignore `DllImport` attribute

**Screenshots**

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 11.0.3

**Additional context**

**Comment by timunie** [comment]
Can you prepare a minimum sample to reproduce?

**Comment by BreyerW** [comment]
@timunie  This is the smallest i could get
[repro.zip](https://github.com/AvaloniaUI/Avalonia/files/12329172/repro.zip)

You will have to start the program to see error. And BTW it was AVLN:00C not AVLOC:00C my bad.

**Comment by DmitryZhelnin** [comment]
Duplicate of #10046

**Comment by timunie** [comment]
It works totally fine if I move the dll import into another class and consume it from there. One may call it Win32Helper for example (just to unblock you). 

@DmitryZhelnin I also think it's a duplicate of #10046 so I'm closing this as duplicate now. 

@BreyerW Thx for the repo btw 👍
